status/reporter_gitlab.go

```diff
@@ -131,6 +131,20 @@ func (r *GitLabReporter) setCommitStatus(report TestReport) error {
 		Description: gitlab.Ptr(report.Summary),
 	}
 
+	// Special case for gitLab `running` state because of a bug where it can't be updated to the same state again
+	if glState == gitlab.Running {
+		allCommitStatuses, _, err := r.client.Commits.GetCommitStatuses(r.projectID, r.sha, nil)
+		if err != nil {
+			return fmt.Errorf("error while getting all comments for merge-request %d: %w", r.mergeRequest, err)
+		}
+		existingCommitStatus := r.GetExistingCommitStatus(allCommitStatuses, report.FullName)
```

Comment:
don't we need a unittest here? it's the aim of this PR

Reply:
Good catch, added the unit test.